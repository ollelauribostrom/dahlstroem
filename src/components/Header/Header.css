.header {
  display: flex;
  justify-content: space-between;
  padding: 0 0 100px 0;
  font-family: 'Roboto', sans-serif;
}

.header__title {
  margin: 0;
  font-weight: 300;
  font-size: 20px;
}

.header__title__link {
  color: hsla(0, 0%, 0%, 0.8);
  text-decoration: none;
}

.menu--mobile {
  display: none;
}

@media only screen and (max-device-width: 1024px) {
  .menu--desktop {
    display: none;
  }
  .menu--mobile {
    display: block;
  }
}
