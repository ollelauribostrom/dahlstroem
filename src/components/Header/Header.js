import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'gatsby';

import Menu from '../Menu';
import Sidebar from '../Sidebar';

const Header = ({ siteTitle, path, color, menuBackground, menuColor }) => (
  <div className="header">
    <div>
      <h1 className="header__title">
        <Link
          to="/"
          className="header__title__link"
          style={{ color: color.hex }}
        >
          {siteTitle}
        </Link>
      </h1>
    </div>
    <div className="menu menu--desktop">
      <Menu
        path={path}
        color={color}
        menuBackground={menuBackground}
        menuColor={menuColor}
      />
    </div>
    <div className="menu menu--mobile">
      <Sidebar
        color={color}
        menuBackground={menuBackground}
        menuColor={menuColor}
      />
    </div>
  </div>
);

Header.propTypes = {
  siteTitle: PropTypes.string.isRequired,
  path: PropTypes.string.isRequired,
  color: PropTypes.shape({ hex: PropTypes.string }),
  menuBackground: PropTypes.shape({ hex: PropTypes.string }),
  menuColor: PropTypes.shape({ hex: PropTypes.string }),
};

Header.defaultProps = {
  color: { hex: '#000' },
};

export default Header;
