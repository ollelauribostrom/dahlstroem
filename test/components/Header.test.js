import React from 'react';
import { shallow } from 'enzyme';

import Header from '../../src/components/Header';

describe('Tests for Header', () => {
  it('renders correctly', () => {
    const wrapper = shallow(
      <Header siteTitle="test" path="/test" color={{ hex: 'fff' }} />
    );
    expect(wrapper).toMatchSnapshot();
  });
  it('uses default value for color if none is provided', () => {
    const wrapper = shallow(<Header siteTitle="test" path="/test" />);
    expect(wrapper).toMatchSnapshot();
  });
});
