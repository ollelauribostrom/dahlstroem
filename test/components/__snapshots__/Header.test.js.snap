// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for Dropdown renders correctly 1`] = `
<div
  className="header"
>
  <div>
    <h1
      className="header__title"
    >
      <mockConstructor
        className="header__title__link"
        style={
          Object {
            "color": "fff",
          }
        }
        to="/"
      >
        test
      </mockConstructor>
    </h1>
  </div>
  <div
    className="menu menu--desktop"
  >
    <Menu
      color={
        Object {
          "hex": "fff",
        }
      }
      path="/test"
    />
  </div>
  <div
    className="menu menu--mobile"
  >
    <Sidebar
      color={
        Object {
          "hex": "fff",
        }
      }
    />
  </div>
</div>
`;

exports[`Tests for Dropdown uses default value for color if none is provided 1`] = `
<div
  className="header"
>
  <div>
    <h1
      className="header__title"
    >
      <mockConstructor
        className="header__title__link"
        style={
          Object {
            "color": "#000",
          }
        }
        to="/"
      >
        test
      </mockConstructor>
    </h1>
  </div>
  <div
    className="menu menu--desktop"
  >
    <Menu
      color={
        Object {
          "hex": "#000",
        }
      }
      path="/test"
    />
  </div>
  <div
    className="menu menu--mobile"
  >
    <Sidebar
      color={
        Object {
          "hex": "#000",
        }
      }
    />
  </div>
</div>
`;
